components/camera/platforms/jetson/camera.go

```diff
@@ -0,0 +1,114 @@
+package jetsoncamera
+
+// #include <sys/utsname.h>
+// #include <string.h>
+import "C"
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime"
+	"unsafe"
+)
+
+// GetOSInformation pulls relevant OS attributes as an OSInformation struct
+// Kernel and Device will be "unkown" if unable to retrieve info from the filesystem
+// returns an error if kernel version or device name is unavailable
+func DetectOSInformation() (OSInformation, error) {
+	kernelVersion, err := getKernelVersion()
+	if err != nil {
+		return OSInformation{}, fmt.Errorf("failed to get kernel version: %w", err)
+	}
+	deviceName, err := getDeviceName()
+	if err != nil {
+		return OSInformation{}, fmt.Errorf("failed to get device name: %w", err)
+	}
+	osInfo := OSInformation{
+		Name:   runtime.GOOS,
+		Arch:   runtime.GOARCH,
+		Kernel: kernelVersion,
+		Device: deviceName,
+	}
+	return osInfo, nil
+}
+
+// getKernelVersion returns the Linux kernel version ($ uname -r)
+func getKernelVersion() (string, error) {
+	var utsName C.struct_utsname
+	if C.uname(&utsName) == -1 {
+		return Unknown, fmt.Errorf("uname information unavailable (%v)", utsName)
+	}
+	release := C.GoString((*C.char)(unsafe.Pointer(&utsName.release[0])))
+	return release, nil
+}
+
+// getDeviceName returns the model name of the device
+// ($ cat /sys/firmware/devicetree/base/model)
+func getDeviceName() (string, error) {
+	devicePath := "/sys/firmware/devicetree/base/model"
+	device, err := os.ReadFile(devicePath)
+	if err != nil {
+		return Unknown, err
+	}
+	return string(bytes.TrimRight(device, "\x00")), nil
+}
+
+// Validate checks if the daughterboard and driver are supported and installed on the device
+func Validate(osInfo OSInformation, daughterboardName string, driverName string) error {
+	board, ok := cameraInfoMappings[osInfo.Device]
+	if !ok {
+		return fmt.Errorf("the %s device is not supported on this platform", osInfo.Device)
+	}
+	daughterboard, ok := board.Daughterboards[daughterboardName]
+	if !ok {
+		return fmt.Errorf("the %s daughterboard is not supported on this platform", daughterboardName)
+	}
+	driver, ok := board.Modules[driverName]
+	if !ok {
+		return fmt.Errorf("the %s driver is not supported on this platform", driverName)
```

Comment:
From @penguinland :

Mild preference to join these two lines together, so it's clear that the current value of `err` won't be used after this if statement:
```go
if err := checkDaughterBoardConnected(daughterboard); err != nil {
    ...
}
```
Same suggestion on line 75.

Reply:
Changed to shorthand.